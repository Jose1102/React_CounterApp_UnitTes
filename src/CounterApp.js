import React, { Fragment, useState }  from 'react';
import PropTypes from 'prop-types';


const CounterApp = ( { value =10 }) => {

  
    const [ counter, setCounter ] = useState( value ); // Retorna un []
   
    // handdleApp : Constante para imprimir el evento del teclado
    const handdleApp = () =>{
        setCounter(counter+1);
    }

    const handdleSubtract = () => {
        setCounter(counter-1);
    }

    const handdleReseat = () => {
        setCounter(value);

    }
    return (
        <Fragment>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={handdleApp}>+1</button>
            <button onClick={handdleReseat}>Reseat</button>
            <button onClick={handdleSubtract}>-1</button>
            

        </Fragment>
    );
}


CounterApp.propTypes = {
    value : PropTypes.number
}

export default CounterApp;

