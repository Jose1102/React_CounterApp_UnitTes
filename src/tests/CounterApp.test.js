import { shallow } from "enzyme"
import CounterApp from '../CounterApp'
describe('deberia hacer la tarea ', ()=>{
    let wraper = shallow(<CounterApp />);

    beforeEach( ()=> {
        wraper = shallow(<CounterApp />)
    })


    test(' debe de mostrar el <CounterApp /> ', () => {
        expect( wraper ).toMatchSnapshot(); 
    })


    test('debe mostrar el valor por defecto que es 100 ', () => {
        const wraper = shallow( <CounterApp value={100}/>)
        const textoCounter = wraper.find( 'h2' ).text().trim();
        //console.log(textoCounter);
        expect( textoCounter ).toBe( '100' );
    })

    test('debe de incrementar con el boton de +1 ', () => {

        const btn1 = wraper.find( 'button' ).at(0).simulate('click');      
        //console.log(btn1.html())  
        const textoCounter = wraper.find( 'h2' ).text().trim();
        expect( textoCounter ).toBe( '11' );

    })


    test('debe de disminuir con el boton de -1 ', () => {

        const btn1 = wraper.find( 'button' ).at(2).simulate('click');      
        //console.log(btn1.html())  
        const textoCounter = wraper.find( 'h2' ).text().trim();
        expect( textoCounter ).toBe( '9' );

    })
    
    test('debe de reiniciar el valor por defecto con el boton Reseat ', () => {

        const wraper = shallow( <CounterApp value={105}/>)
        const btn1 = wraper.find( 'button' ).at(0).simulate('click');
        const textoCounter = wraper.find( 'h2' ).text().trim();
        console.log(textoCounter)
        const btnReseat = wraper.find( 'button' ).at(1).simulate('click');
        const textoNuevo = wraper.find( 'h2' ).text().trim();
        console.log("este es "+ textoNuevo)
        expect( textoNuevo ).toBe( '105' )

        
    })
    
    
})