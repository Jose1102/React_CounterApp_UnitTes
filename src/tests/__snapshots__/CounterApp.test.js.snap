// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deberia hacer la tarea   debe de mostrar el <CounterApp />  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reseat
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
